cli/src/commands/git/colocation.rs

```diff
@@ -0,0 +1,334 @@
+// Copyright 2025 The Jujutsu Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use std::io::Write as _;
+
+use jj_lib::file_util::IoResultExt as _;
+use jj_lib::git;
+use jj_lib::op_store::RefTarget;
+use jj_lib::repo::Repo as _;
+
+use crate::cli_util::CommandHelper;
+use crate::command_error::CommandError;
+use crate::command_error::user_error;
+use crate::command_error::user_error_with_message;
+use crate::commands::git::maybe_add_gitignore;
+use crate::git_util::is_colocated_git_workspace;
+use crate::ui::Ui;
+
+/// Show the current colocation status
+#[derive(clap::Args, Clone, Debug)]
+pub struct GitColocationStatusArgs {}
+
+/// Enable colocation (convert into a colocated Jujutsu/Git repository)
+///
+/// This moves the underlying Git repository that is found inside the .jj
+/// directory to the root of the Jujutsu workspace. This allows you to
+/// use Git commands directly in the Jujutsu workspace.
+#[derive(clap::Args, Clone, Debug)]
+pub struct GitColocationEnableArgs {}
+
+/// Disable colocation (convert into a non-colocated Jujutsu/Git repository)
+///
+/// This moves the Git repository that is at the root of the Jujutsu
+/// workspace into the .jj directory. Once this is done you will no longer
+/// be able to use Git commands directly in the Jujutsu workspace.
+#[derive(clap::Args, Clone, Debug)]
+pub struct GitColocationDisableArgs {}
+
+/// Manage Jujutsu repository colocation with Git
+#[derive(clap::Subcommand, Clone, Debug)]
+pub enum GitColocationCommand {
+    Status(GitColocationStatusArgs),
+    Enable(GitColocationEnableArgs),
+    Disable(GitColocationDisableArgs),
+}
+
+pub fn cmd_git_colocation(
+    ui: &mut Ui,
+    command: &CommandHelper,
+    subcommand: &GitColocationCommand,
+) -> Result<(), CommandError> {
+    let workspace_command = command.workspace_helper(ui)?;
+
+    // Make sure that the repository supports git colocation commands
+    // before running any subcommand
+    repo_supports_git_colocation_commands(workspace_command)?;
```

Comment:
I meant this can be moved to `cmd_git_colocation_*()` where `workspace_command` is loaded again.